From b1f6eac2792f5d774b0a4b30cea0da28d11bf08b Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Mon, 24 Sep 2012 11:26:49 -0400
Subject: [PATCH 38/42] Coverity issue 13091

Bug Description:  Local var array "mychange" can go out of scope

Fix Description:  Declare "mychange" at the top of the function so it does not go out of scope.

Reviewed by: nkinder(Thanks!)
(cherry picked from commit d58ae20b75ccae99b2122a4a0d4fcf773a51115a)
---
 ldap/servers/slapd/plugin_acl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/slapd/plugin_acl.c b/ldap/servers/slapd/plugin_acl.c
index 3bc3f21..eebc292 100644
--- a/ldap/servers/slapd/plugin_acl.c
+++ b/ldap/servers/slapd/plugin_acl.c
@@ -136,6 +136,7 @@ plugin_call_acl_mods_update ( Slapi_PBlock *pb, int optype )
 	struct slapdplugin	*p;
 	int					rc = 0;
 	void				*change = NULL;
+	void				*mychange[2];
 	Slapi_Entry			*te = NULL;
 	Slapi_DN			*sdn = NULL;
 	Operation			*operation;
@@ -158,7 +159,6 @@ plugin_call_acl_mods_update ( Slapi_PBlock *pb, int optype )
 		break;
 	  case SLAPI_OPERATION_MODRDN:
 	  {
-		void *mychange[2];
 		char *newrdn = NULL;
 		Slapi_DN *psdn = NULL;
 		char *pdn = NULL;
-- 
1.8.1.4

